### ISSUE TITLE
LooseVersion import doesn't work on ansible < 2.9.20 and < 2.10.8

### ISSUE THREAD

**Issue by evgeni** [theforeman/foreman-ansible-modules#1352]
##### SUMMARY

Originally found by @igramic and reported to me via chat.

When using Ansible 2.9.19 and older, the change from #1347 makes our helper fail to load and thus the whole collection unusable:

```console
% ansible -m theforeman.foreman.domain -a 'server_url=https://localhost username=admin password=changeme' localhost
localhost | FAILED! => {
    "msg": "Could not find imported module support code for foreman_helper.  Looked for either LooseVersion.py or version.py"
}
```

Whereas on 2.9.20 and newer, this correctly loads and executes:
```console
% ansible -m theforeman.foreman.domain -a 'server_url=https://localhost username=admin password=changeme' localhost
localhost | FAILED! => {
    "changed": false,
    "msg": "missing required arguments: name"
}
```

To the best of my knowledge, this was fixed in 2.9.20 via https://github.com/ansible/ansible/commit/f8caa43dc7a7beaf66e719e1e5f7bc6ff70fa7e6

The same applies to the 2.10 branch for [2.10.0,2.10.7], but the error is slightly different:
```console
% ansible -m theforeman.foreman.domain -a 'server_url=https://localhost username=admin password=changeme' localhost
localhost | FAILED! => {
    "msg": "Could not find imported module support code for ansible_collections.theforeman.foreman.plugins.modules.domain.  Looked for (['ansible.module_utils.compat.version.LooseVersion', 'ansible.module_utils.compat.version'])"
}
```

2.10.8 contains https://github.com/ansible/ansible/commit/f51852557dffc2ce5f501c1ab4986e867f55fec07

and then 2.11+ have the "original" fix https://github.com/ansible/ansible/commit/3e1f6484d77f2d7546952cfa22a8534d74ed3dc6

I think the main question is: do we care? And if we do, how should we adjust to not run into this again in future?
Keep in mind, that to "fix" the currently present issue, we'd need to revert the change from #1347 which means we'll see linting errors on Python 3.10+ and with Python 3.12 stuff will stop work until we re-implement the change (thus, again, dropping support for these versions).

For comparison, the various OSes (supported by Foreman) and their Ansible versions:
* Debian 10 (Buster): 2.9.16
* Debian 11 (Bullseye): 2.10.8
* Ubuntu 20.04 (Focal): 2.9.6
* EL7 (EPEL): 2.9.27
* EL7 (cfgmgmt SIG): 2.9.27
* EL8 (cfgmgmt SIG): 2.9.27
* EL8 (Stream): 2.12.2

##### ISSUE TYPE
 - Bug Report

##### ANSIBLE VERSION

```
ansible 2.9.19
  config file = /tmp/aold/lol/ansible.cfg
  configured module search path = ['/home/evgeni/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /tmp/aold/lib64/python3.5/site-packages/ansible
  executable location = /tmp/aold/bin/ansible
  python version = 3.5.10 (default, Sep 17 2021, 00:00:00) [GCC 11.2.1 20210728 (Red Hat 11.2.1-1)]
```

##### COLLECTION VERSION

```
3.2.0
```

**Comment by evgeni** [comment]
cc @ehelms @mdellweg for visibility

**Comment by ehelms** [comment]
Is #1347 just a nice to have? This looks like it only affects the two Debian versions? (and how close are we to deprecating them?)

**Comment by rpurcelldesj** [comment]
We are experiencing the problem here. Our tower template using the foreman collection aren't working anymore using Ansible 2.9.2, we are switching back to the v3.1.0 of the collection untill further notice.

**Comment by evgeni** [comment]
@rpurcelldesj why is your tower still running 2.9.2?

**Comment by rpurcelldesj** [comment]
@evgeni I do agree it should be more up to date but unfortunately for now it is what it is. We are in the process of upgrading to automation platform and we should fix this and stay up to date in the near future..

**Comment by evgeni** [comment]
@ehelms right now, its "nice to have as it eliminates a warning when you run sanity tests on python 3.10" (watch https://github.com/theforeman/foreman-ansible-modules/pull/1353 if anything will actually fail).

Yes, it affects Debian 10 Buster and Ubuntu 20.04 Focal, we're close to drop Buster, but we need to first gain support Ubuntu 22.04 (which isn't released yet) before we can drop that.

And there are obviously users who didn't update to the latest version of Ansible 2.9.z (or 2.10, but that never really got much use I think), even tho it was probably available to them (like Rene here)

**Comment by evgeni** [comment]
```
ERROR: Found 70 import issue(s) on python 3.10 which need to be resolved:
ERROR: plugins/inventory/foreman.py:172:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
ERROR: plugins/module_utils/foreman_helper.py:28:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
ERROR: plugins/modules/activation_key.py:197:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives (at plugins/module_utils/foreman_helper.py:28:0)
ERROR: plugins/modules/architecture.py:100:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives (at plugins/module_utils/foreman_helper.py:28:0)
ERROR: plugins/modules/auth_source_ldap.py:191:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives (at plugins/module_utils/foreman_helper.py:28:0)
```

**Comment by rpurcelldesj** [comment]
Hi guy's, just a followup on my part, does the PR https://github.com/theforeman/foreman-ansible-modules/pull/1353 have some chance to get merged or the change https://github.com/theforeman/foreman-ansible-modules/commit/5dc0f47d7feeb8b2624d89360635ff40c297ccbf is here to stay ?

Thanks

**Comment by evgeni** [comment]
@rpurcelldesj we're trying to restore compatibility with those old versions.

**Comment by evgeni** [comment]
3.3.0 just hit Galaxy, and besides me forgetting to add a changelog entry for this, it has the fix you require :)
